module/VuFind/src/VuFind/ILS/Driver/PAIA.php

```diff
@@ -1785,9 +1835,39 @@ protected function paiaLogin($username, $password)
                 self::SCOPE_WRITE_ITEMS . " " .
                 self::SCOPE_CHANGE_PASSWORD
         ];
-        $responseJson = $this->paiaPostRequest('auth/login', $post_data);
 
-        $responseArray = $this->paiaParseJsonAsArray($responseJson);
+        // perform full PAIA auth and get patron info
+        try {
+            $result = $this->httpService->post(
+                $this->paiaURL . 'auth/login',
+                json_encode($post_data),
+                'application/json; charset=UTF-8',
+                $this->paiaTimeout,
+                $header_data
+            );
+        } catch (\Exception $e) {
+            throw new ILSException($e->getMessage());
```

Comment:
Should you pass along the exception code as well as the message?

Reply:
I have removed this try-catch block in reference to #1319. A more in depth exception-handling should not be the scope of this pull request.